### PR TITLE
Add attribute whitespace exemptions

### LINKED ISSUE #259: collapseAttributeWhitespace should skip attributes where whitespace is known to be significant
In most attributes, whitespace can safely be collapsed, but there are a handful of attributes where whitespace is significant. Skimming through the [attribute index](https://html.spec.whatwg.org/multipage/indices.html#attributes-3), here are some I found that have visible effect:

- `title`: moderately popular, in fact, for manual line breaks and simple alignment in tooltips. This is the one I started filing this issue for, because I was actively using it on a page.
- `alt` (with `white-space: pre`)
- `label` (probably especially with `white-space: pre`)
- `pattern`
- `value`
- `placeholder`
- All event handler content attributes (`on*`), which should be getting JS minification so this should then leave them alone too.

Other attributes that *may* be affected:
- `content` (definitely should be in theory),
- `download` (kinda should be but probably isn’t, I imagine all browsers will normalise whitespace as part of sanitising the file name).

This *may* be the complete set.

I propose that `collapseAttributeWhitespace` should skip attributes of these eight identified names, plus names starting with `on`.

### PR DESCRIPTION
**Pull request by j9t**
Resolves #259

## Summary by CodeRabbit

* **Bug Fixes**
  * Preserve meaningful attribute whitespace: spaces in regexes, pre-filled form values, visible labels/placeholders, and inside event-handler string literals are no longer collapsed.

* **Documentation**
  * Updated acknowledgements to include an additional contributor and broadened wording.

* **Chores**
  * Bumped package version to 6.2.1 and added corresponding changelog entry.

### CODE REVIEW

File: src/lib/attributes.js [hunk 1]
```diff
@@ -299,7 +299,7 @@ function hasAttrName(name, attrs) {
 function cleanAttributeValue(tag, attrName, attrValue, options, attrs, minifyHTMLSelf) {
   // Apply early whitespace normalization if enabled
   // Preserves special spaces (no-break space, hair space, etc.) for consistency with `collapseWhitespace`
-  if (options.collapseAttributeWhitespace) {
+  if (options.collapseAttributeWhitespace && attrName !== 'pattern' && attrName !== 'value') {
```
[the remark this answers is not in the document]
Comment (j9t) [reply src/lib/attributes.js]: Would push back (please do, too, if this would miss something important). Machine-supported explanation:

The additional exemptions aren't warranted. Collapsing whitespace in `title` and `alt` is safe: screen readers normalize whitespace universally, and browsers render both attribute values under default `white-space: normal`, so no output changes. The one edge case for `title`—newline-separated tooltip lines in Firefox—is an inherent trade-off of the option, and like any other case where `collapseAttributeWhitespace` is too aggressive, it can always be turned off or `htmlmin:ignore` used selectively. Screen readers normalize whitespace in alt text; browsers normalize placeholder rendering; social platforms and search engines normalize meta content. The `white-space: pre` argument for `alt` and `label` is theoretical to the point of irrelevance—alt text renders as image fallback or is processed by assistive technology, neither of which applies `white-space: pre` to the raw attribute string. `download` filenames are browser-normalized. For `on*` handlers, these already receive unconditional `trimWhitespace` at line 313 regardless of `collapseAttributeWhitespace`; the one edge case—a `//` comment followed by code on the next line with `minifyJS: false`—is real but contrived, and `minifyJS` is the correct tool for inline JS whitespace.

File: tests/html.spec.js [hunk 2]
```diff
@@ -4360,6 +4360,18 @@ describe('HTML', () => {
...
+    input = '<input pattern="\\d  \\d">';
+    assert.strictEqual(await minify(input, { collapseAttributeWhitespace: true }), input);
+    input = '<input pattern="  [a-z]+">';
+    assert.strictEqual(await minify(input, { collapseAttributeWhitespace: true }), input);
+
+    // `value` whitespace may be intentional (pre-filled form field contents), so it must not be touched
+    input = '<input value="  hello  ">';
+    assert.strictEqual(await minify(input, { collapseAttributeWhitespace: true }), input);
+    input = '<input value="foo  bar">';
+    assert.strictEqual(await minify(input, { collapseAttributeWhitespace: true }), input);
```
[the remark this answers is not in the document]
Comment (j9t) [reply tests/html.spec.js:4373]: [See earlier note.](https://github.com/j9t/html-minifier-next/pull/267#discussion_r3168860348)

File: src/lib/attributes.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (j9t) [reply src/lib/attributes.js]: I think you were right on `title`, `placeholder`, `on*`—PR updated once more. Will close here, but re-review and correct/challenge where needed.

File: tests/html.spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (j9t) [reply tests/html.spec.js:4373]: I think you were right on `title`, `placeholder`, `on*`—PR updated once more. Will close here, but re-review and correct/challenge where needed.

File: tests/html.spec.js [hunk 3]
```diff
@@ -4337,29 +4336,66 @@ describe('HTML', () => {
...
+
+    // `placeholder` whitespace is significant (leading spaces are visually rendered in inputs)
+    input = '<input placeholder="  enter name">';
+    assert.strictEqual(await minify(input, { collapseAttributeWhitespace: true }), input);
+    input = '<input placeholder="first  last">';
+    assert.strictEqual(await minify(input, { collapseAttributeWhitespace: true }), input);
+
+    // Event handler whitespace is significant (spaces inside JS string literals must not be collapsed)
+    input = '<button onclick="alert(\'→     ←\')">x</button>';
+    assert.strictEqual(await minify(input, { collapseAttributeWhitespace: true }), input);
```
[the remark this answers is not in the document]
Comment (j9t) [reply tests/html.spec.js:4398]: Done!
